sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_file_system_client_async.py

```diff
@@ -97,48 +103,110 @@ def __init__(
+
+        self._api_version = get_api_version(kwargs)
+        self._client = self._build_generated_client(self.url)
+        self._datalake_client_for_blob_operation = self._build_generated_client(self._container_client.url)
         self._loop = kwargs.get('loop', None)
 
-    async def __aexit__(self, *args):
+    def _build_generated_client(self, url: str) -> AzureDataLakeStorageRESTAPI:
+        client = AzureDataLakeStorageRESTAPI(
+            url,
+            base_url=url,
+            file_system=self.file_system_name,
+            pipeline=self._pipeline
+        )
+        client._config.version = self._api_version  # pylint: disable=protected-access
+        return client
+
+    def _format_url(self, hostname: str) -> str:
+        return _format_url(self.scheme, hostname, self.file_system_name, self._query_str)
+
+    async def __aexit__(self, *args: Any) -> None:
         await self._container_client.close()
         await self._datalake_client_for_blob_operation.close()
         await super(FileSystemClient, self).__aexit__(*args)
 
-    async def close(self):
-        # type: () -> None
-        """ This method is to close the sockets opened by the client.
+    async def close(self) -> None:
+        """This method is to close the sockets opened by the client.
         It need not be used when using with a context manager.
         """
         await self.__aexit__()
 
+    @classmethod
+    def from_connection_string(
+        cls, conn_str: str,
+        file_system_name: str,
+        credential: Optional[Union[str, Dict[str, str], "AzureNamedKeyCredential", "AzureSasCredential", "AsyncTokenCredential"]] = None,  # pylint: disable=line-too-long
+        **kwargs: Any
+    ) -> Self:
+        """
+        Create FileSystemClient from a Connection String.
+
+        :param str conn_str:
+            A connection string to an Azure Storage account.
+        :param file_system_name: The name of file system to interact with.
+        :type file_system_name: str
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string,
+            an instance of a AzureSasCredential or AzureNamedKeyCredential from azure.core.credentials,
+            an account shared access key, or an instance of a TokenCredentials class from azure.identity.
+            Credentials provided here will take precedence over those in the connection string.
+            If using an instance of AzureNamedKeyCredential, "name" should be the storage account name, and "key"
+            should be the storage account key.
+        :type credential:
+            ~azure.core.credentials.AzureNamedKeyCredential or
+            ~azure.core.credentials.AzureSasCredential or
+            ~azure.core.credentials.TokenCredential or
```

Comment:
Same feedback here as other decouple PR, just comb for async and make necessary changes